lib/application.js

```diff
@@ -284,7 +285,10 @@ app.handler = function(type, options) {
   var handler = this._handlers[type] = remotes.handler(type, options);
 
   remotes.classes().forEach(function(sharedClass) {
-    sharedClass.ctor.emit('mounted', app, sharedClass, remotes);
+    /* HACK!!! */
+    if (sharedClass.name !== 'ctrl') {
```

Comment:
I think there are two options:

1) Don't emit `mounted` for controllers that don't provide EventEmitter mixed into the constructor. (BTW this is rather unusual, to emit events on a constructor - a Function instance. Typically, events are emitted on object instances created by a constructor function.) Instead of checking for controller name, check whether the constructor provides `emit` method.

```js
if (sharedClass.ctor && sharedClass.ctor.emit) {
  sharedClass.ctor.emit('mounted', app, sharedClass, remotes);
}
```

2) Automatically add EventEmitter when a controller is registered via `app.controller`. See [this code](https://github.com/strongloop/loopback-datasource-juggler/blob/2189b9a746071b599d6a129313b2cdd11052e36b/lib/model-builder.js#L184-L206) in juggler for an example, just ignore the deprecation warnings.

```js
app.controller = function(ctor, options) {
  if (!ctor.emit) {
    // mix in EventEmitter into the constructor (not the instances!)
    var events = new EventEmitter();
    for (var f in EventEmitter.prototype) {
      if (typeof EventEmitter.prototype[f] === 'function') {
        ctor[f] = EventEmitter.prototype[f].bind(events);
      }
    }
  }

  var self = this;
  // etc.
};
```

---

Personally, I am proposing to use the first approach and don't emit `mounted` method for controllers. Here are my arguments:

 - It's less code/functionality to maintain.
 - We can always add it later.
 - Controllers work differently from models. There is always a new controller instance created for each request and this instance receives `app` in the constructor. There is no need to pass the `app` object via `mounted` event.

@richardpringle @ritch thoughts?